usb.go

```diff
@@ -187,16 +187,15 @@ func (c *Context) OpenDevices(opener func(desc *DeviceDesc) bool) ([]*Device, er
 	var ret []*Device
 	for _, dev := range list {
 		desc, err := c.libusb.getDeviceDesc(dev)
+		defer c.libusb.dereference(dev)
 		if err != nil {
-			c.libusb.dereference(dev)
 			reterr = err
 			continue
 		}
 
 		if opener(desc) {
```

Comment:
please change this to:
if !opener(desc) {
  continue
}
and unindent the rest